qiling/os/memory.py

```diff
@@ -63,6 +63,29 @@ def __read_string(self, addr: int) -> str:
     def __write_string(self, addr: int, s: str, encoding: str):
         self.write(addr, bytes(s, encoding) + b'\x00')
 
+    def __getitem__(self, key: Union[slice, int]) -> bytearray:
+        if isinstance(key, slice):
+            start = key.start
+            stop = key.stop
+            step = key.step
+
+            if step is not None and step != 1:
+                # step != 1 means we have to do copy, don't allow it.
+                raise IndexError("Only support slicing continous memory")
+
+            return self.ql.mem.read(start, stop - start)
+        elif isinstance(key, int):
+            return self.ql.mem.read(key, 1)
+        else:
+            raise KeyError("Wrong type of key")
+    
+    def __setitem__(self, key: Union[slice, int], value: Union[bytes, bytearray]):
+        if isinstance(key, int):
+            self.ql.mem.write(key, value)
+        else:
+            # Slicing doesn't make sense in writing.
```

Comment:
That actually could make sense if we followed the `bytearray` semantics.
Following it would not be entirely possible, since it allows expanding and deleting slots - which we can't have with memory, but we can still overwrite the given slice area with the given bytes; if the slice size doesn't match the length of the data, data could be cropped if larger, or repeated if smaller.

Reply:
cropped is accepted while repeat might lead to performance overhead.